client/src/hooks/UserContext.tsx

```diff
@@ -28,14 +35,48 @@ export const UserProvider = ({ children }: Props) => {
     onOpen: () => console.log('websocket opened'),
     shouldReconnect: () => false,
     onMessage: ({ data }) => {
-      if (!user) return;
       const end_time = new Date();
-      const start_time = new Date(JSON.parse(data).data.start_time);
+      if (!user) return;
+      const message = JSON.parse(data);
+      const start_time = new Date(message.data.start_time);
       const latency = Number(end_time) - Number(new Date(start_time));
       reportLatency(user.id, latency);
     },
   });
 
+  const measureThroughput = async () => {
+    try {
+      const file = await fetch(throughput_test);
+      const blob = await file.blob();
+
+      const formData = new FormData();
+      formData.append('file', blob);
+      formData.append('start_time', String(new Date().getTime()));
+      formData.append('size', String(blob.size));
+      const res = await fetch(`${BASE_URL}/throughput`, {
```

Comment:
Why fetch?

Reply:
Did not manage to serialize the response to form data using axios